### ISSUE TITLE
Use "native" vectorisation

### ISSUE THREAD

**Issue by Bisaloo** [epiverse-trace/numberize#11]
Opening this issue as a reminder of the discussions started in https://github.com/epiverse-trace/numberize/pull/6#discussion_r1532548560.

Most R functions are vectorized by default (using loops in C) so it may be much more efficient to rely on this rather than a `vapply()` loop.

**Comment by bahadzie** [comment]
You are right that most of the functions I was using were already vectorized.

The only one I had an issue with was [match() in numberize.R](https://github.com/epiverse-trace/numberize/blob/fc3282563c056b6f7cbbae51d2fb0fcd5916af35/R/numberize.R#L72). It didn't natively vectorize it as I expected and it might be an issue in my understanding of how it works. The only way I got it to work was to use `apply()` which felt like kicking the can down the road. That's why I didn't fully vectorize it natively.

After the CRAN release, I will see if I can come up with a solution for it.

**Comment by Bisaloo** [comment]
Partially fixed by #15
